parser-java/kvs-archived-media-rekognition-label-detection-sample/src/main/java/com/amazonaws/kinesisvideo/utilities/H264ImageDetectionBoundingBoxRenderer.java

```diff
@@ -0,0 +1,143 @@
+                        Optional<FragmentMetadataVisitor.MkvTagProcessor> tagProcessor) throws FrameProcessException {
+
+        BufferedImage bufferedImage = decodeH264Frame(frame, trackMetadata);
+
+        /* Only send key frames to Rekognition */
+        if (sampleRate == 0) {
+            if (frame.isKeyFrame()) {
+                renderFrame(bufferedImage);
+            }
+        }
+        else {
+            /* Only send to Rekognition every N frames */
+            if ((frameNumber % sampleRate) == 0) {
+                renderFrame(bufferedImage);
+            }
+            frameNumber++;
+        }
+    }
+
+    protected void renderFrame(final BufferedImage bufferedImage) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+        try {
+            ImageIO.write(bufferedImage, "png", outputStream);
+            ByteBuffer imageBytes = ByteBuffer.wrap(outputStream.toByteArray());
+            List<BoundingBox> boundingBoxes = new ArrayList<>();
+            long startTime = System.nanoTime();
+            sendToRekognition(imageBytes, bufferedImage, boundingBoxes);
+            long endTime = System.nanoTime();
+            long totalTime = endTime - startTime;
+            double seconds = (double)totalTime / 1_000_000_000.0;
+            log.info("Time to Rekognize frame: " + seconds + " seconds");
+            log.info("----------------------");
+
+            for (BoundingBox boundingBox: boundingBoxes) {
+                addBoundingBoxToImage(bufferedImage, boundingBox);
+            }
+            kinesisVideoFrameViewer.update(bufferedImage);
+        }
+        catch (IOException e) {
+            log.warn("Error with png conversion", e);
+        }
+    }
+
+    protected void sendToRekognition(ByteBuffer imageBytes, BufferedImage bufferedImage, List<BoundingBox> boundingBoxes) {
+        DetectLabelsRequest request = new DetectLabelsRequest()
+                .withImage(new Image()
+                        .withBytes(imageBytes))
+                .withMaxLabels(10)
+                .withMinConfidence(77F);
+
+        try {
+            DetectLabelsResult result = rekognitionClient.detectLabels(request);
+            List<Label> labels = result.getLabels();
+
+            int width = bufferedImage.getWidth();
+            int height = bufferedImage.getHeight();
+
+            log.info("Detected Labels:");
+            for (Label label: labels) {
+                log.info(label.getName() + ": " + label.getConfidence().toString());
+            }
+
+            for (Label label: labels) {
+                for (Instance instance: label.getInstances()) {
+                    boundingBoxes.add(instance.getBoundingBox());
+                    final int left = (int) (instance.getBoundingBox().getLeft() * width);
+                    final int top = (int) (instance.getBoundingBox().getTop() * height);
+                    addTextToImage(bufferedImage, label.getName(), left, top);
```

Comment:
It seems that the text rendering process is still here

Reply:
Ok I got rid of the bounding boxes array list and simply added bounding boxes on the spot. The function is now named sendToRekognition and it adds text and bounding boxes.